mobile/lib/pages/article_selection_page.dart

```diff
@@ -22,146 +22,106 @@ class ArticleSelectionPage extends StatefulWidget {
-        return Padding(
-          padding: const EdgeInsets.only(bottom: 40),
-          child: GridView.builder(
-            controller: _scrollController,
-            padding: const EdgeInsets.all(10),
-            gridDelegate: const SliverGridDelegateWithFixedCrossAxisCount(
-              crossAxisCount: 2,
+    return Padding(
+      padding: const EdgeInsets.only(bottom: 50), // Helps with the player not covering the last article
+      child: GridView.builder(
+        controller: scrollController,
+        padding: const EdgeInsets.all(20), // Space around the grid
+        gridDelegate: const SliverGridDelegateWithFixedCrossAxisCount(
+          crossAxisCount: 2,
+          crossAxisSpacing: 20, // Space between columns
+          mainAxisSpacing: 20, // Space between rows
+        ),
+        itemCount: articles.value.length,
+        itemBuilder: (context, index) {
+          final article = articles.value.elementAt(index);
+          return GestureDetector(
+            onTap: () => context.push(
+              RaRoutes.articleId(article.id),
+              extra: article,
             ),
-            itemCount: _articles.length,
-            itemBuilder: (context, index) {
-              final article = _articles.elementAt(index);
-              return Padding(
-                padding:
-                    const EdgeInsets.symmetric(horizontal: 15, vertical: 15),
-                child: GestureDetector(
-                  onTap: () => context.push(
-                    RaRoutes.articleId(article.id),
-                    extra: article,
-                  ),
-                  child: ColorShadowedCard2(
-                    shadowColor: shadowColor(context, index),
-                    footer: DefaultTextStyle(
-                      style: context.textStyles.textSmall.copyWith(
-                        color: context.colors.highlightGreen,
-                      ),
-                      child: Padding(
-                        padding: const EdgeInsets.all(5),
-                        child: HtmlWidget(article.title),
-                      ),
-                    ),
-                    child: Image.network(
-                      article.thumbnail,
-                      errorBuilder: (context, error, stackTrace) => AspectRatio(
-                        aspectRatio: 1,
-                        child: Center(
-                          child: Image.asset('assets/defaultMedia.png'),
-                        ),
-                      ),
+            child: ColorShadowedCard2(
+              shadowColor: shadowColor(context, index),
+              footer: DefaultTextStyle(
+                style: context.textStyles.textSmall.copyWith(
+                  color: context.colors.highlightGreen,
+                ),
+                child: Padding(
+                  padding: const EdgeInsets.all(5),
+                  child: HtmlWidget(article.title),
+                ),
+              ),
+              child: Image.network(
```

Comment:
Consider adding loading animation (with RaProgressIndicator from `lib/components/utility/ra_progress_indicator.dart`) to these images with `loadingBuilder` attribute.

Reply:
I implemented it tho I am not sure it actually works, needs checking, the images are loading too fast